state.go

```diff
@@ -573,19 +585,75 @@ func NewUpdateErrorState(err menderError, update UpdateResponse) State {
 }
 
 func (ue *UpdateErrorState) Handle(ctx *StateContext, c Controller) (State, bool) {
-	// TODO error handling, repeat status update for as long as needed
-	c.ReportUpdateStatus(ue.update, statusFailure)
-
-	// TODO fetch logs from logger
-	c.UploadLog(ue.update, []LogEntry{
-		LogEntry{
-			Timestamp: time.Now().Format(time.RFC3339),
-			Level:     "error",
-			Message:   "update failed",
-		},
-	})
+	return NewUpdateStatusReportState(ue.update, statusFailure), false
+}
+
+// Wrapper for mandatory update state reporting, the state handler will attempt
```

Comment:
` Wrapper for mandatory update status reporting. The state handler will attempt`